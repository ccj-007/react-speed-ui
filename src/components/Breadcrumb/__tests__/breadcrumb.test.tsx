import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Breadcrumb, BreadcrumbItem } from '../../index'

describe('test badge component', () => {
  it('basic use should render', () => {
    const { asFragment } = render(
      <>
        <BreadcrumbItem>Home</BreadcrumbItem>
        <BreadcrumbItem>
          <a href="https://github.com/ccj-007/react-speed-ui">Application Center</a>
        </BreadcrumbItem>
        <BreadcrumbItem>
          <a href="https://github.com/ccj-007/react-speed-ui">Application List</a>
        </BreadcrumbItem>
        <BreadcrumbItem>An Application</BreadcrumbItem>
        <Breadcrumb allSeparator={<span>-{">"}</span>}>
          <BreadcrumbItem>Home</BreadcrumbItem>
          <BreadcrumbItem>
            <a href="https://github.com/ccj-007/react-speed-ui">Application Center</a>
          </BreadcrumbItem>
          <BreadcrumbItem>
            <a href="https://github.com/ccj-007/react-speed-ui">Application List</a>
          </BreadcrumbItem>
          <BreadcrumbItem>An Application</BreadcrumbItem>
        </Breadcrumb>
        <Breadcrumb>
          <BreadcrumbItem disabled>Home</BreadcrumbItem>
          <BreadcrumbItem disabled separator={<span>---{">"}</span>}>
            <a href="https://github.com/ccj-007/react-speed-ui">Application Center</a>
          </BreadcrumbItem>
          <BreadcrumbItem disabled separator={<span>-----{">"}</span>}>
            <a href="https://github.com/ccj-007/react-speed-ui">Application List</a>
          </BreadcrumbItem>
          <BreadcrumbItem disabled separator={<span>-----{">"}</span>}>
            An Application
          </BreadcrumbItem>
        </Breadcrumb>
      </>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})