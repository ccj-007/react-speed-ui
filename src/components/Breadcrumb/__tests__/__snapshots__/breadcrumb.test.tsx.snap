// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test badge component basic use should render 1`] = `
<DocumentFragment>
  <div
    class="speed-breadcrumbItem"
  >
    <span
      class=""
    >
      Home
    </span>
  </div>
  <div
    class="speed-breadcrumbItem"
  >
    <span
      class=""
    >
      <a
        href="https://github.com/ccj-007/react-speed-ui"
      >
        Application Center
      </a>
    </span>
  </div>
  <div
    class="speed-breadcrumbItem"
  >
    <span
      class=""
    >
      <a
        href="https://github.com/ccj-007/react-speed-ui"
      >
        Application List
      </a>
    </span>
  </div>
  <div
    class="speed-breadcrumbItem"
  >
    <span
      class=""
    >
      An Application
    </span>
  </div>
  <div
    class="speed-breadcrumb"
  >
    <div
      class="speed-breadcrumbItem"
    >
      <span
        style="margin: 0px 10px;"
      >
        <span>
          -&gt;
        </span>
      </span>
      <span
        class=""
      >
        Home
      </span>
    </div>
    <div
      class="speed-breadcrumbItem"
    >
      <span
        style="margin: 0px 10px;"
      >
        <span>
          -&gt;
        </span>
      </span>
      <span
        class=""
      >
        <a
          href="https://github.com/ccj-007/react-speed-ui"
        >
          Application Center
        </a>
      </span>
    </div>
    <div
      class="speed-breadcrumbItem"
    >
      <span
        style="margin: 0px 10px;"
      >
        <span>
          -&gt;
        </span>
      </span>
      <span
        class=""
      >
        <a
          href="https://github.com/ccj-007/react-speed-ui"
        >
          Application List
        </a>
      </span>
    </div>
    <div
      class="speed-breadcrumbItem"
    >
      <span
        style="margin: 0px 10px;"
      >
        <span>
          -&gt;
        </span>
      </span>
      <span
        class=""
      >
        An Application
      </span>
    </div>
  </div>
  <div
    class="speed-breadcrumb"
  >
    <div
      class="speed-breadcrumbItem"
    >
      <span
        style="margin: 0px 10px;"
      />
      <span
        class="speed-breadcrumbItem-disabled"
      >
        Home
      </span>
    </div>
    <div
      class="speed-breadcrumbItem"
    >
      <span>
        ---&gt;
      </span>
      <span
        class="speed-breadcrumbItem-disabled"
      >
        <a
          class="speed-breadcrumbItem-disabled"
          href="javascript:;"
        >
          Application Center
        </a>
      </span>
    </div>
    <div
      class="speed-breadcrumbItem"
    >
      <span>
        -----&gt;
      </span>
      <span
        class="speed-breadcrumbItem-disabled"
      >
        <a
          class="speed-breadcrumbItem-disabled"
          href="javascript:;"
        >
          Application List
        </a>
      </span>
    </div>
    <div
      class="speed-breadcrumbItem"
    >
      <span>
        -----&gt;
      </span>
      <span
        class="speed-breadcrumbItem-disabled"
      >
        An Application
      </span>
    </div>
  </div>
</DocumentFragment>
`;
